### DISCUSSION TITLE
Strapi custom controller is not decrypting a record when accessed from /graphql

### DISCUSSION THREAD

**Discussion by nabhakumthekar** [strapi/strapi#7774] [Legacy Topics]
Hello,
I am not able to fetch decrypted value for record when I try to access strapi from graphql.
For Example,
I am storing user critical information as username and email in an encrypted value to mariaDB. 
To encrypt and decrypt this value I was able to override Users.js controller and it works flawless.
I have a dataview which returns me some information related to dataview and user information of that dataview.
When I access this dataview via end points exposed by strapi such as **http://localhost:8080/dataview/1** I get decrypted value of user fields in response.
But, When I try to access same dataview via graphql query such as  **http://localhost:8080/graphql** decryption is not working 
query is 
```
query {
  dataviews {
    id
    user_creator {
      email
    }
    user_owner {
      email
    }
  }
}
```
and my controller code for find method is
```
async find(ctx, __populate) {
    let entities = await strapi.services.dataview.find({
      "user_owner.id": ctx.state.user.id,
    });

    entities.forEach((entity) => {
      if (entity.user_owner) {
        decryptUser(entity.user_owner);
      }
      if (entity.user_creator) {
        decryptUser(entity.user_creator);
      }
    });

    return entities.map((entity) =>
      sanitizeEntity(entity, { model: strapi.models.dataview })
    );
  }
```
Here, If I print entities in console.log I see decrypted values but when this controller returns a response on postman or UI values are still encrypted. 
I am not able to find root cause for this issue. Any help is appreciated.

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.
